pkg/graph/resolvers/base_struct.go

```diff
@@ -0,0 +1,29 @@
+package resolvers
+
+import (
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/applychanges"
+	"github.com/cmsgov/easi-app/pkg/authentication"
+	"github.com/cmsgov/easi-app/pkg/models"
+	"github.com/cmsgov/easi-app/pkg/storage"
+)
+
+// ApplyChangesAndMetaData is a hook to call before updating a baseStruct object in the database it will
+// 1. Update the base struct meta data
+// 2. Apply changes from the changes object to the provided IBaseStruct
+func ApplyChangesAndMetaData(logger *zap.Logger, bs models.IBaseStruct, changes map[string]interface{}, principal authentication.Principal, store *storage.Store) error {
```

Comment:
It doesn't look this currently needs to take `logger` or `store` as a parameter; are there future needs for those anticipated, or can we remove them? If it's just about having them for the future, I'd almost rather just pass around the whole Context.